### ISSUE TITLE
dbt fails to run on duckdb==0.4.0

### ISSUE THREAD

**Issue by jecolvin** [jwills/dbt-duckdb#9]
Any dbt command (I personally tried `dbt seed` and `dbt run`) fails with the following error:

```
Runtime Error
  TransactionContext Error: cannot commit - no transaction is active
```

Downgrading to duckdb v0.3.2 resolves the issue.

**Comment by jwills** [comment]
Well that's lame, I will take a look shortly as I was hoping to do the upgrade to 0.4.0 in the next version!

**Comment by dbeatty10** [comment]
I got the same error message as @jecolvin with duckdb 0.4.0.

Downgrading to duckdb 0.3.4 worked again.

There's some interesting discussion about this error message here:
https://github.com/duckdb/duckdb/issues/3395

I'm guessing some default implementation(s) within dbt-core are adding `commit`s that duckdb 0.4.0 can't handle.

_Maybe_ [here](https://github.com/dbt-labs/dbt-core/blob/3e3ecb1c3f49efcfee9dbd09374bcf23e4e5b565/core/dbt/adapters/sql/connections.py#L133-L134)?

A handful of other possibilities come up via a grep or [search in GitHub](https://github.com/dbt-labs/dbt-core/search?q=commit) (but there's a bit of noise to sort through).

**Comment by jwills** [comment]
So FWIW the new testing framework makes it really simple to see where things are going wrong here:

```
/usr/local/lib/python3.9/site-packages/dbt/tests/fixtures/project.py:389: in create_test_schema
    self.adapter.create_schema(relation)
/usr/local/lib/python3.9/site-packages/dbt/adapters/sql/impl.py:163: in create_schema
    self.commit_if_has_connection()
/usr/local/lib/python3.9/site-packages/dbt/adapters/base/impl.py:191: in commit_if_has_connection
    self.connections.commit_if_has_connection()
/usr/local/lib/python3.9/site-packages/dbt/adapters/base/connections.py:274: in commit_if_has_connection
    self.commit()
/usr/local/lib/python3.9/site-packages/dbt/adapters/sql/connections.py:158: in commit
    self.add_commit_query()
/usr/local/lib/python3.9/site-packages/dbt/adapters/sql/connections.py:134: in add_commit_query
    return self.add_query("COMMIT", auto_begin=False)
/usr/local/lib/python3.9/site-packages/dbt/adapters/sql/connections.py:70: in add_query
    cursor.execute(sql, bindings)
```

Basically @dbeatty10 it looks like the problem is that dbt will run a commit after _any_ create/drop schema operation, regardless of whether or not the connection actually has an open transaction; I'm honestly not sure why that is the right thing to do for other adapter, but I suspect I could fix it by overriding the impl of that method to check the state of the connection before issuing the commit, so that's what I'm going to try out.

**Comment by jwills** [comment]
Ugh, okay-- finally figured out what was going wrong here and everything is now fixed as of `dbt-duckdb` version 1.1.3!
